middle_end/flambda2/to_cmm/to_cmm.ml

```diff
@@ -454,6 +454,22 @@ let unary_primitive env dbg f arg =
          environment, then [Env.closure_offset] might return [None], even though
          the set of closures has been seen by [To_cmm_closure]. *)
       None, C.unreachable)
+  | Is_boxed_float ->
+    (* As a note, this omits the [Is_in_value_area] check that exists in
+       [caml_make_array], which is used by non-Flambda 2 compilers. *)
+    ( None,
+      Cmm.Cifthenelse
```

Comment:
It'd be great if this could use the `C.ite` helper function; it reasonably increases readability I find. More generally, `to_cmm.ml` should not need to use `Cmm` directly (except in very rare cases), and instead use the smart constructors provided by the `C` module (i.e. by either `Cmm_helpers` or `To_cmm_helpers`).

Reply:
Fixed